### ISSUE TITLE
security_socket_sendmsg not exist and can't attach kprobe

### ISSUE THREAD

**Issue by ChenLingPeng** [pixie-io/pixie#932]
**Describe the bug**
when kernel build with CONFIG_SECURITY_NETWORK not set, security_socket_sendmsg is inlined. 
In this situation, socket_tracer will not work

```
I20230301 07:05:24.697929 26969 bcc_wrapper.cc:170] Initializing BPF program ...
I20230301 07:05:33.953138 26969 scoped_timer.h:48] Timer(init_bpf_program) : 9.26 s
cannot attach kprobe, probe entry may not exist
W20230301 07:05:37.417877 26969 stirling.cc:411] Source Connector (registry name=socket_tracer) not instantiated, error: Internal : Unable to attach kprobe for security_socket_sendmsg using probe_entry_security_socket_sendmsg
```

using sock_sendmsg maybe another option

**To Reproduce**

**Expected behavior**

**Screenshots**
If applicable, add screenshots to help explain your problem. Please make sure the screenshot does not contain any sensitive information such as API keys or access tokens.

**Logs**
Please attach the logs by running the following command:
```
./px collect-logs
```

**App information (please complete the following information):**
- Pixie version
- K8s cluster version
- Node Kernel version
- Browser version

**Additional context**

**Comment by ddelnano** [comment]
Hi @ChenLingPeng, thanks for reporting this issue and opening up a PR to address it!

I was able to verify that the functions we probe are dependent on `CONFIG_SECURITY_NETWORK` (as you mentioned). I believe we need to add probes to the entry and return of `sock_sendmsg`, however, it was intentional that we are tracing `security_sock_sendmsg`.

In Google's [Container optimized OS](https://cloud.google.com/container-optimized-os/docs), we've noticed that it is possible for the kernel to call call `security_socket_sendmsg` rather than `socket_sendmsg` (https://github.com/pixie-io/pixie/commit/508686ff248f9210bc34bc4580dd371a7b3072a7). If we change which function we probe, we will miss the probe on COS in some cases.

I believe we will need to do something more dynamic such as inspecting the kernel config or available probes to determine the correct function to attach to. Let me discuss with the rest of the team to get consensus, but I'm fairly confident that's what the ideal change would do.

**Comment by ddelnano** [comment]
@ChenLingPeng I've confirmed with the team that we will need a dynamic solution for this so that we ensure the COS case is still handled properly.

Can you give more detail on what kernel you are using? Is this custom built or are you using an upstream kernel?

**Comment by ChenLingPeng** [comment]
Thanks @ddelnano for the comments.
I'm using a custom build kernel.

**Comment by ChenLingPeng** [comment]
Hi @ddelnano , in which branch of [Container optimized OS](https://cloud.google.com/container-optimized-os/docs) will call  `security_socket_sendmsg` rather than `sock_sendmsg` ? I've checkout [the source code](https://cos.googlesource.com/third_party/kernel/) and find that `security_socket_sendmsg/security_socket_recvmsg` is only called by `sock_sendmsg/sock_recvmsg`

**Comment by ddelnano** [comment]
@ChenLingPeng a person familiar with the background on this change explained that this happens when AppArmor is enabled on Container optimized OS. We never identified why AppArmor (or another LSM) causes this behavior, but switching to the `security_` prefixed functions fixed the tracing bug we were investigating at the time.

Regarding options for dynamically attaching the probe, I was thinking that identifying what kernel symbols are available (via `/proc/kallsyms` or something similar) would be a good mechanism for determine what function to probe.

**Comment by ChenLingPeng** [comment]
@ddelnano thanks for the details. Looking forward to the new fix

**Comment by ddelnano** [comment]
@ChenLingPeng do you have interest in working on that change? Happy to provide support if you want to take that on.

Otherwise we can look into prioritizing this, but I don't have an estimate for when that will be.

**Comment by ChenLingPeng** [comment]
@ddelnano sorry, I'm not familiar with C++

**Comment by ddelnano** [comment]
This was completed in #1811 and is available in the v0.14.10 vizier release.
